types/vote.go

```diff
@@ -419,25 +417,32 @@ func SignAndCheckVote(
 		return true, err
 	}
 	vote.Signature = v.Signature
-	vote.ExtensionSignature = nil
-	vote.Timestamp = v.Timestamp
-
-	if extensionsEnabled {
-		if vote.Type != PrecommitType {
-			panic(fmt.Sprintf("Extensions are enabled, but vote is not precommit (vote type: %T)", vote.Type))
-		}
-
-		if len(v.ExtensionSignature) == 0 {
-			return false, ErrVoteNoSignature // Non-recoverable because the vote is malformed.
-		}
 
-		if vote.BlockID.IsNil() {
-			return false, ErrVoteExtSignatureForNilBlock // Non-recoverable because the vote is malformed.
-		}
+	isPrecommit := vote.Type == PrecommitType
+	if !isPrecommit && extensionsEnabled {
+		// Non-recoverable because the caller passed parameters that don't make sense
+		return false, fmt.Errorf("only Precommit votes may have extensions enabled; vote type: %T", vote.Type)
```

Comment:
Same here.

Reply:
I would print the entire vote here. It is evident that if it is not a Precommit, it must be a Prevote. Otherwise, signing will fail.